src/common/snippets/src/lowered/pass/extract_loop_invariants.cpp

```diff
@@ -0,0 +1,204 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/lowered/pass/extract_loop_invariants.hpp"
+
+#include "snippets/itt.hpp"
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/snippets_isa.hpp"
+#include "snippets/utils.hpp"
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+namespace {
+void remove_last_loop_id(const std::shared_ptr<Expression>& expr) {
+    auto loop_ids = expr->get_loop_ids();
+    OPENVINO_ASSERT(!loop_ids.empty(), "Expr loop_ids should not be empty when remove last loop id.");
+    loop_ids.pop_back();
+    expr->set_loop_ids(loop_ids);
+}
+
+size_t get_stride_after_move_outer(const LoopPort& loop_port) {
+    const auto& expr_port = loop_port.expr_port;
+    const auto& shape = expr_port->get_descriptor_ptr()->get_shape();
+    size_t shape_dim_idx = utils::get_dim_idx(*expr_port, loop_port.dim_idx);
+
+    size_t stride = 1;
+    for (size_t i = shape_dim_idx; i < shape.size(); ++i) {
+        if (utils::is_dynamic_value(shape[i])) {
+            return utils::get_dynamic_value<size_t>();
+        }
+        stride *= shape[i];
+    }
+    return stride;
+}
+}  // namespace
```

Comment:
Why some functions are in this unnamed namespace, but others are static and private. Isn't it essentially the same thing? Why don't we move all static private methods to this namespace?

Reply:
all moved to unnamed namespace, Thanks Ivan.